File: code/packages/python/haskell-parser/tests/test_haskell_parser.py
from haskell_parser import create_haskell_parser, parse_haskell


def test_parser_root_is_file() -> None:
    ast = parse_haskell("x")
    assert ast.rule_name == "file"


def test_create_parser_returns_working_parser() -> None:
    parser = create_haskell_parser("let { x = y } in x", "2010")
    ast = parser.parse()
    assert ast.rule_name == "file"

File: code/packages/lua/haskell_parser/tests/test_haskell_parser.lua
local haskell_parser = require("coding_adventures.haskell_parser")

describe("haskell_parser", function()
    it("uses file as the root rule", function()
        local ast = haskell_parser.parse("x")
        assert.are.equal("file", ast.rule_name)
    end)

    it("parses explicit-brace let expressions", function()
        local ast = haskell_parser.parse("let { x = y } in x", "2010")
        assert.are.equal("file", ast.rule_name)
    end)
end)
